### ISSUE TITLE
Error!!! No Channel Info in /users/PAE57595_pass_barcode01_83605eaf_16.fast5

### ISSUE THREAD

**Issue by chenchen-eng** [WGLab/DeepMod#39]
Hello, teacher, how should this problem be solved？

**Comment by liuqianhn** [comment]
It seems that there is no channel info in your fast5 files. Please show the output of "h5ls -r /users/PAE57595_pass_barcode01_83605eaf_16.fast5 | head -n 50" so that I can comment more.

**Comment by chenchen-eng** [comment]
> 您的fast5文件中似乎没有频道信息。请显示“ h5ls -r /users/PAE57595_pass_barcode01_83605eaf_16.fast5 | head -n 50”的输出，以便我进一步评论。
Thank you, teacher. My first 50 lines are as follows. How should I deal with the original FAST5 file?
/                        Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Basecall_1D_000 Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Basecall_1D_000/BaseCalled_template Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Basecall_1D_000/BaseCalled_template/Fastq Dataset {SCALAR}
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Basecall_1D_000/Summary Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Basecall_1D_000/Summary/basecall_1d_template Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Segmentation_000 Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Segmentation_000/Summary Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Analyses/Segmentation_000/Summary/segmentation Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Raw Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/Raw/Signal Dataset {163018/Inf}
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/channel_id Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/context_tags Group
/read_001009ec-891e-4377-86d2-06ceab0cb2ad/tracking_id Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4 Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Basecall_1D_000 Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Basecall_1D_000/BaseCalled_template Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Basecall_1D_000/BaseCalled_template/Fastq Dataset {SCALAR}
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Basecall_1D_000/Summary Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Basecall_1D_000/Summary/basecall_1d_template Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Segmentation_000 Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Segmentation_000/Summary Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Analyses/Segmentation_000/Summary/segmentation Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Raw Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/Raw/Signal Dataset {165995/Inf}
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/channel_id Group
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/context_tags Group, same as /read_001009ec-891e-4377-86d2-06ceab0cb2ad/context_tags
/read_0029e9f8-d867-4b18-a727-481ada57a2f4/tracking_id Group, same as /read_001009ec-891e-4377-86d2-06ceab0cb2ad/tracking_id
/read_002b4830-62da-4e62-a54a-4ae609d95007 Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Basecall_1D_000 Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Basecall_1D_000/BaseCalled_template Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Basecall_1D_000/BaseCalled_template/Fastq Dataset {SCALAR}
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Basecall_1D_000/Summary Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Basecall_1D_000/Summary/basecall_1d_template Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Segmentation_000 Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Segmentation_000/Summary Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Analyses/Segmentation_000/Summary/segmentation Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Raw Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/Raw/Signal Dataset {61491/Inf}
/read_002b4830-62da-4e62-a54a-4ae609d95007/channel_id Group
/read_002b4830-62da-4e62-a54a-4ae609d95007/context_tags Group, same as /read_001009ec-891e-4377-86d2-06ceab0cb2ad/context_tags
/read_002b4830-62da-4e62-a54a-4ae609d95007/tracking_id Group, same as /read_001009ec-891e-4377-86d2-06ceab0cb2ad/tracking_id
/read_00435e94-11d6-40b5-b5e2-561c9f9ad45e Group
/read_00435e94-11d6-40b5-b5e2-561c9f9ad45e/Analyses Group
/read_00435e94-11d6-40b5-b5e2-561c9f9ad45e/Analyses/Basecall_1D_000 Group
/read_00435e94-11d6-40b5-b5e2-561c9f9ad45e/Analyses/Basecall_1D_000/BaseCalled_template Group

**Comment by liuqianhn** [comment]
@chenchen-eng Your fast5 files have multi-fast5 format, and the tool is not supported yet. You can use [ont_fast5_api](https://github.com/nanoporetech/ont_fast5_api) to convert them to single-fast5 format before using the tools. We are working on the improvement of the tool including multi-fast5 format supporting, but the improvement is not available in a short time.

**Comment by chenchen-eng** [comment]
Thank you, can We use ont_fast5_API?The teacher?Mine is a fungus sample. Can your model be used directly?

**Comment by chenchen-eng** [comment]
teacher，How should this error be resolved?Error!!! No Fastq data in /users/xulab/lida/m6a/01_PH-1/20200810_1005_4G_PAE88034_798dbc59/output/71/8016b946-83a0-4db4-bc82-780698a2be82.fast5

**Comment by chenchen-eng** [comment]
/                        Group
/Analyses                Group
/Analyses/Basecall_1D_000 Group
/Analyses/Basecall_1D_000/BaseCalled_template Group
/Analyses/Basecall_1D_000/BaseCalled_template/Fastq Dataset {SCALAR}
/Analyses/Basecall_1D_000/Summary Group
/Analyses/Basecall_1D_000/Summary/basecall_1d_template Group
/Analyses/Basecall_1D_001 Group
/Analyses/Basecall_1D_001/BaseCalled_template Group
/Analyses/Basecall_1D_001/BaseCalled_template/Fastq Dataset {SCALAR}
/Analyses/Basecall_1D_001/BaseCalled_template/ModBaseProbs Dataset {6886, 6}
/Analyses/Basecall_1D_001/BaseCalled_template/Move Dataset {35312}
/Analyses/Basecall_1D_001/BaseCalled_template/Trace Dataset {35312, 8}
/Analyses/Basecall_1D_001/Summary Group
/Analyses/Basecall_1D_001/Summary/basecall_1d_template Group
/Analyses/Segmentation_000 Group
/Analyses/Segmentation_000/Summary Group
/Analyses/Segmentation_000/Summary/segmentation Group
/Analyses/Segmentation_001 Group
/Analyses/Segmentation_001/Summary Group
/Analyses/Segmentation_001/Summary/segmentation Group
/Raw                     Group
/Raw/Reads               Group
/Raw/Reads/Read_9383     Group
/Raw/Reads/Read_9383/Signal Dataset {72404/Inf}
/UniqueGlobalKey         Group
/UniqueGlobalKey/channel_id Group
/UniqueGlobalKey/context_tags Group
/UniqueGlobalKey/tracking_id Group
no events data?teacher

**Comment by liuqianhn** [comment]
1. You can find ont_fast5_API via the link I provided. I think, you already install and use it.
2. If there are errors such as `No Fastq` or `no events`, what you can do is:
Use `h5ls -r ` to check the path in fasta5 files as you did. In fact, in your fast5 file, you have two basecalled branch: `Basecall_1D_000` and `Basecall_1D_001`. You need to make sure that the path is correct to find fastq and event via `--basecall_1d Basecall_1D_000` or `--basecall_1d Basecall_1D_001`. Here, `--basecall_1d Basecall_1D_000` is the default option. In your example fast5 file, the default option does not work since your `Basecall_1D_000` in this file does not have complete results.
3. If `no events` happens, you also need to use `h5ls -r ` to check the path in fasta5 files as you did. In your example file, it seems that you have `move` table rather than `events`, thus, you need to `--move True` to tell the tool to find `move` table rather than `events` table.
In short, adding `--basecall_1d Basecall_1D_000 --move True` to your running command should solve the issue you have. 
Please note that the available well-trained model is not updated, and I have worked on the improvement. But the improvement cannot be available in a short time.
